source/blueprints/user/routes.py

```diff
@@ -128,12 +128,13 @@ def change_password():
 def user_del():
     """Allow the user to delete his own account"""
 
-    data = request.json
-
     try:
-        delete_confirm = UserDeleteDTO().load(data)
-        delete_user(delete_confirm["confirm_deletion"])
-        db.session.commit()
+        if not current_user.deleted_at:
+            delete_user()
+            db.session.commit()
+            
+        else:   
+            return 'User already deleted.'
```

Comment:
Just a small tweak here,  replace `return 'User already deleted.'` with the following code:

```
abort(409, {"type": CauseTypeError.DATA_VIOLATION_ERROR.value, "data": "User already deleted."})
```

Note that I choose the **409 status code**  - which stands for **CONFLICT**.

In this app, usually when we are sending a **successful response** we are going to use the `DataResponse` class, and when we are sending a **error response** we are going to use the `abort()` function.

Let me know if you have any questions about it.